kernel/include/lapic.h

```diff
@@ -0,0 +1,151 @@
+// https://habr.com/en/post/446312/ Has a nice graphical overview of APICs
+// https://wiki.osdev.org/APIC
+
+#pragma once
+#include <stdint.h>
+
+// Stolen from https://docs.rs/x86/0.21.0/src/x86/xapic.rs.html#76
+#pragma region XAPIC REGISTERS
+#define XAPIC_ID 0x020
```

Comment:
Use a packed struct instead of these defines, you can even mark members as const to make the compiler enforce their read only status.